torch/csrc/generic/Tensor.cpp

```diff
@@ -907,23 +906,45 @@ static THIndexTensor* THPTensor_(_calculateLinearIndices)(
   }
   THLongStorage_free(indexerSize);
 
+#ifdef THC_GENERIC_FILE
+  // Call GPU kernel for index calculation
+  THCudaLongTensor *cudaIndices =
+    THCudaLongTensor_newWithSize1d(LIBRARY_STATE indexingElements);
+  long baseOffset = THTensor_(storageOffset)(LIBRARY_STATE indexed);
+
+  // Need to pass broadcast Tensors to API, pass NULL ptr for all empty
+  // (i.e. not-advanced indexed) dims
+  THCudaLongTensor *indexers[THTensor_(nDimension)(LIBRARY_STATE indexed.get())];
```

Comment:
You can't use variable length arrays. They are a GCC extension and they already broke torch7 builds on Windows. Just use an `std::vector` and `.reserve()` the memory